sympy/solvers/diophantine.py

```diff
@@ -1184,6 +1189,92 @@ def diop_DN(D, N, t=symbols("t", integer=True)):
                 return sol
 
 
+def special_diop_DN(D, N):
+    """
+    Solves the equation `x^2 - Dy^2 = N` for the special case where
+    `1 < N**2 < D` and `D` is not a perfect square.
+    It is better to call `diop_DN` rather than this function, as
+    the former checks the condition `1 < N**2 < D`, and calls the latter only
+    if appropriate.
+
+    Usage
+    =====
+
+    ``special_diop_DN(D, N)``: D and N are integers as in `x^2 - Dy^2 = N`.
+
+    Details
+    =======
+
+    ``D`` and ``N`` correspond to D and N in the equation.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.diophantine import special_diop_DN
+    >>> special_diop_DN(13, -3) # Solves equation x**2 - 13*y**2 = -3
+    [(7, 2), (137, 38)]
+
+    The output can be interpreted as follows: There are two fundamental
+    solutions to the equation `x^2 - 13y^2 = -3` given by (7, 2) and
+    (137, 38). Each tuple is in the form (x, y), i.e solution (7, 2) means
+    that `x = 7` and `y = 2`.
+
+    >>> special_diop_DN(2445, -20) # Solves equation x**2 - 2445*y**2 = -20
+    [(445, 9), (17625560, 356454), (698095554475, 14118073569)]
+
+    See Also
+    ========
+
+    diop_DN()
+
+    References
+    ==========
+
+    .. [1] Quadratic Diophantine Equations, T. Andreescu and D. Andrica,
+        Springer, 2015.
+    """
+    assert (1 < N**2 < D) and (not integer_nthroot(D, 2)[1])
```

Comment:
We generally avoid bare asserts (though I think this sort of assertion is clear and self explanatory). It can be recast as a ValueError:

```python
if not (1 < N**2 < D):
    if N**2 == 1:
        raise ValueError("N must be > 1")
    else:
        raise ValueError("N must be less than %s" % D)
if integer_nthroot(D, 2)[1]:
    raise ValueError("D must not be a perfect square")
```

BUT...I would recommend making this a private method by prefixing its name with an underscore with the understanding that this should be called directly from the `diop_DN` which should also check that D and N are integers and has confirmed that D is not a perfect square. Then if anyone calls it on their own and gets stuck in a loop it is their own fault and you dn't penalize the function with these redundant checks.